### ISSUE TITLE
`enforce-canonical-classes` rule misses various issues reported by `suggestCanonicalClasses`

### ISSUE THREAD

**Issue by mcmxcdev** [schoero/eslint-plugin-readable-tailwind#307]
### Description

After running `eslint --fix` including `enforce-canonical-classes` rule multiple times across the whole codebase, there are various issues reported by the TailwindCSS extension directly that are not flagged by the `enforce-canonical-classes` rule.

This is one example of what the TailwindCSS extension complains about, e.g. "The class `text-(--text-primary)` can be written as `text-text-primary`"

When hovering over it, this is what Intellisense shows me:
```
.text-\(--text-primary\) {
  color: var(--text-primary);
}
```

After applying the suggested fix, when hovering over it, its still a valid class name:

```
.text-text-primary {
  color: var(--text-primary);
}
```

Therefore, I would expect this necessary fix to be detected and applied by the `enforce-canonical-classes` rule too. 

### Flavor

TSX

### Code Input

```tsx
`text-(--text-primary)`
```

### Expected Behavior

```tsx
`text-(--text-primary)` gets turned into `text-text-primary`
```

### Actual Behavior

```tsx
`text-(--text-primary)` -> TailwindCSS extension complains that it can be rewritten, but `enforce-canonical-classes` doesn't.
```

### ESLint Log

```shell
.
```

### ESLint Config

```typescript
import eslintPluginBetterTailwindcss from "eslint-plugin-better-tailwindcss";

...

{
    plugins: {
      "better-tailwindcss": eslintPluginBetterTailwindcss,
    },
    settings: {
      "better-tailwindcss": {
        entryPoint: resolve(__dirname, "../config-tailwindcss/base.css"),
      },
    },
  },
  {
    rules: {
      "better-tailwindcss/enforce-canonical-classes": "error",
    },
  },
```

### Versions

- ESLint: 9.39.2
- Parser: tseslint.parser 
- Plugin: ?
- Node: v24.13.0

**Comment by ragnsan** [comment]
Any way to sponsor someone to work on this specific thing? Super high value!
